pkg/engine/task/podexec/pod_exec.go

```diff
@@ -182,20 +188,22 @@ func untarFile(fs afero.Fs, r io.Reader, fileName string) error {
 			if err != nil {
 				return err
 			}
+			defer f.Close() // nolint
```

Comment:
Isn't this now running into the issue you're trying to avoid, i.e. each file close will wait until all operations have completed? This can be avoided by explicitly adding scope through braces:
```go
case tar.TypeReg:
		{
			f, err := fs.OpenFile(target, os.O_CREATE|os.O_RDWR, os.FileMode(header.Mode))
			if err != nil {
				return err
			}
			defer f.Close() // nolint

			// copy over contents
			if _, err := io.Copy(f, tr); err != nil {
				return err
			}
		}
```

Reply:
Wait, looks like I'm wrong and defers are scoped differently than I thought, see https://goplay.space/#MnGumtleKY3. Well, in that case, closing files once all operations have been completed is better than leaking file descriptors.

Reply:
Shouldn't matter - I only untar one file.